packages/modelserver-client/src/model-server-client-api-v2.ts

```diff
@@ -0,0 +1,133 @@
+/*********************************************************************************
+ * Copyright (c) 2022 STMicroelectronics and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
+ * available at https://opensource.org/licenses/MIT.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR MIT
+ *********************************************************************************/
+import { Operation } from 'fast-json-patch';
+
+import { ServerConfiguration, SubscriptionOptions } from './model-server-client-api-v1';
+import { Model, ModelServerMessage } from './model-server-message';
+import { ModelServerElement } from './model/base-model';
+import { ModelServerCommand } from './model/command-model';
+import { Diagnostic } from './model/diagnostic';
+import { SubscriptionListener } from './subscription-listener';
+import { AnyObject, TypeGuard } from './utils/type-util';
+
+/** JSON formats supported by the V2 client API. */
+export type JsonFormat = 'json' | 'json-v2';
+
+/** Message formats supported by the V2 client API. */
+export type Format = 'xml' | JsonFormat;
+
+/**
+ * Basic client API to interact with a model server that conforms to the Modelserver API Version 2.
+ */
+export interface ModelServerClientApiV2 {
```

Comment:
I'm not sure. Most of the interfaces are not duplicated (at the moment), as V1 and V2 are still 95% similar (The command execution and notifications parts are the only big differences).

Splitting namespaces might introduce additional inconsistencies or complexity, that we don't necessarily need (or want). For example, we'd have a "Common" (or empty) namespace for APIs shared by V1 and V2, and then "V1" and "V2" specific namespaces. What if we need to specialize one more API for V2? Would we move it from "common" to "v1" and duplicate it in "v2"? Would that break existing V1/V2 clients?

And duplicating everything "just in case" is something we want to avoid (See eclipse-emfcloud/emfcloud-modelserver/pull/169)

Reply:
I see, I think for now having the classes as they are is fine for now. I just was interested to see what the reasoning behind it was. Of course, in the long run we may need to find a better approach if we want to support multiple versions but that is yet to be seen. Since both v1 and v2 still work as intended, this is fine for me.